content/slice/language-guide/exception.md

```diff
@@ -76,29 +76,39 @@ exception TranslationException {
 ```
 
 ```csharp
-public partial class TranslationException
-    : SliceException
+public partial class TranslationException :
+    SliceException
 {
-    public TranslationErrorCode ErrorCode;
-    public string? DetectedLanguage;
+    public TranslationErrorCode ErrorCode { get; set; }
+    public string? DetectedLanguage { get; set; }
+
+    // Parameterless constructor
+    public TranslationException()
+    {
+    }
 
     // Primary constructor
     public TranslationException(
         TranslationErrorCode errorCode,
-        string? detectedLanguage,
-        string? message = null,
-        System.Exception? innerException = null)
-        : base(message, innerException)
+        string? detectedLanguage)
+    {
+        this.ErrorCode = errorCode;
+        this.DetectedLanguage = detectedLanguage;
+    }
+
+    // Secondary constructor
+    public TranslationException(
+        TranslationErrorCode errorCode)
     {
-        ...
+        this.ErrorCode = errorCode;
     }
 }
 ```
 
 {% /aside %}
 
-The mapped C# class provides a primary constructor with parameters for all its fields, plus an optional message and an
-optional inner exception (like most exceptions in C#).
+The mapped C# class provides a parameterless constructor and a primary constructor with parameters for all its fields.
+A class with nullable fields also provides a secondary constructor that sets only the non-nullable fields.
```

Comment:
In the class version, we talk about fields with optional type, instead of nullable fields.

> If any field has an optional type, the mapped class ...

Would be best to be consistent and use the same terminology in both places.

Reply:
Fixed